src/lib/router.ts

```diff
@@ -76,12 +77,49 @@ router.post('/matches/:matchId/highlights', async (request, response, next) => {
   }
 });
 
-router.get('/matches', async (_request, response, next) => {
+type PaginatedMatches = {
+  info: {
+    total: number;
+    itemsPerPage: number;
+    page: number;
+  };
+  results: Match[];
+};
+
+function createMatchesQuery(requestQuery: ParsedQs): Filter<Match> {
+  const query: Filter<Match> = {};
+  if (typeof requestQuery.username === 'string') {
+    query.username = requestQuery.username;
+  }
+  const gameId = Number(requestQuery.gameId);
+  if (gameId) {
+    query.gameId = gameId;
+  }
+  return query;
+}
+
+router.get('/matches', async (request, response, next) => {
   try {
-    const matchesCollection = await getMatchesCollection();
-    const matches = await matchesCollection.find({}).toArray();
+    const page = Number(request.query.page) || 1;
+    const itemsPerPage = Number(request.query.itemsPerPage) || 10;
+    const matchesCollection = getMatchesCollection();
+    const query: Filter<Match> = createMatchesQuery(request.query);
```

Comment:
The type declaration `Filter<Match>` isn't necessary, because we know that `getMatchesCollection()` returns only that type.